control-plane/connect-inject/controllers/endpointsv2/endpoints_controller.go

```diff
@@ -254,13 +256,21 @@ func getServicePorts(service corev1.Service) []*pbcatalog.ServicePort {
 	ports := make([]*pbcatalog.ServicePort, 0, len(service.Spec.Ports)+1)
 
 	for _, p := range service.Spec.Ports {
-		ports = append(ports, &pbcatalog.ServicePort{
-			VirtualPort: uint32(p.Port),
-			//TODO: If the value is a number, infer the correct name value based on
-			// the most prevalent endpoint subset for the port (best-effot, inspect a pod).
-			TargetPort: p.TargetPort.String(),
-			Protocol:   common.GetPortProtocol(p.AppProtocol),
-		})
+		// Service mesh only supports TCP as the L4 Protocol (not to be confused w/ L7 AppProtocol).
+		//
+		// This check is necessary to deduplicate VirtualPort values when multiple declared ServicePort values exist
+		// for the same port, which is possible in K8s when e.g. multiplexing TCP and UDP traffic over a single port.
+		//
+		// If we otherwise see repeat port values in a K8s service, we pass along and allow Consul to fail validation.
+		if p.Protocol == "TCP" {
```

Comment:
nit: wondering if there's some constant in k8s libraries for this?

Reply:
Ah, somehow missed [this](https://github.com/kubernetes/kubernetes/blob/7c42715e57979a1c3cc065f3e3914c0ef3caf779/pkg/apis/core/types.go#L762) the first time I checked! Updated, good callout.